task-sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -877,6 +879,11 @@ def run(
         assert ti.task is not None
         assert isinstance(ti.task, BaseOperator)
 
+    def _on_kill(signum, frame):
+        ti.task.on_kill()
+
+    signal.signal(signal.SIGTERM, _on_kill)
```

Comment:
Nit, but an important one:

```suggestion
    def _on_term(signum, frame):
        ti.task.on_kill()

    signal.signal(signal.SIGTERM, _on_term)
```

Reply:
I wonder if we need to put some `pid` checks in here, to make sure that we don't run this in any subprocesses that the task might fork. What did we do in 2.x?

Reply:
fixed, its checking pid :)